e2e/internal/keyexec/keyexec.go

```diff
@@ -0,0 +1,244 @@
+func RecoverSecretKeyring(t *testing.T) {
+	backupSypgp := filepath.Join(sypgp.DirPath(), "secret-keyring-backup")
+
+	if err := os.Remove(sypgp.SecretPath()); err != nil {
+		t.Fatalf("Unable to remove secret keyring: %v", err)
+	}
+	if err := os.Rename(backupSypgp, sypgp.SecretPath()); err != nil {
+		t.Fatalf("Unable to rename secret keyring: %v", err)
+	}
+}
+
+// RemoveSecretKeyring will delete your secret keyring.
+func RemoveSecretKeyring(t *testing.T) {
+	err := os.Remove(sypgp.SecretPath())
+	if err != nil {
+		t.Fatalf("Unable to remove secret keyring: %v", err)
+	}
+}
+
+// ImportKey will import a key from kpath.
+func ImportKey(t *testing.T, kpath string) ([]byte, error) {
+	e2e.LoadEnv(t, &testenv)
+
+	argv := []string{"key", "import", kpath}
+	execKey := exec.Command(testenv.CmdPath, argv...)
+
+	return execKey.CombinedOutput()
+}
+
+// ImportPrivateKey will take a private key file (kpath) and import it.
+func ImportPrivateKey(t *testing.T, kpath string) (string, []byte, error) {
+	e2e.LoadEnv(t, &testenv)
+
+	//c, err := expect.NewConsole(expect.WithStdout(os.Stdout))
+	c, err := expect.NewConsole(expect.WithStdin(os.Stdin))
+	if err != nil {
+		panic(err)
+	}
+	defer c.Close()
+
+	exportCmd := []string{"key", "import", kpath}
+
+	cmd := exec.Command(testenv.CmdPath, exportCmd...)
+	cmd.Stdin = c.Tty()
+
+	buf := bytes.NewBuffer(nil)
+	cmd.Stderr = buf
+	cmd.Stdout = buf
+
+	go func() {
+		c.ExpectEOF()
+	}()
+
+	err = cmd.Start()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Send the passcode to singularity
+	c.Send("e2etests\n")
+	c.Send("e2etests\n")
+	c.Send("e2etests\n")
+
+	err = cmd.Wait()
+	cm := fmt.Sprintf("%s %s", testenv.CmdPath, strings.Join(exportCmd, " "))
+
+	return cm, buf.Bytes(), err
+}
+
+// ExportPrivateKey will import a private key from kpath.
+func ExportPrivateKey(t *testing.T, kpath, num string, armor bool) (string, []byte, error) {
+	e2e.LoadEnv(t, &testenv)
+
+	//c, err := expect.NewConsole(expect.WithStdout(os.Stdout))
+	c, err := expect.NewConsole(expect.WithStdin(os.Stdin))
+	if err != nil {
+		panic(err)
+	}
+	defer c.Close()
+
+	exportCmd := []string{"key", "export", "--secret"}
+
+	if armor {
+		exportCmd = append(exportCmd, "--armor")
+	}
+
+	exportCmd = append(exportCmd, kpath)
+
+	outErr := bytes.NewBuffer(nil)
+
+	cmd := exec.Command(testenv.CmdPath, exportCmd...)
+	//cmd.Stdin = c.Tty()
+	//cmd.Stdin = outErr
+	cmd.Stdin = c.Tty()
+
+	cmd.Stderr = outErr
+	cmd.Stdout = outErr
+
+	go func() {
+		c.ExpectEOF()
+	}()
+
+	err = cmd.Start()
+	if err != nil {
+		t.Fatalf("unable to run command: %v", err)
+	}
+
+	c.Send(num)
+	c.Send("e2etests\n")
+
+	err = cmd.Wait()
+	cm := fmt.Sprintf("%s %s", testenv.CmdPath, strings.Join(exportCmd, " "))
+
+	return cm, outErr.Bytes(), err
+}
+
+// CorruptKey will take a ASCII key (kpath) and change some chars in it (corrupt it).
+func CorruptKey(t *testing.T, kpath string) {
```

Comment:
Would it be better to use `sed` and call a `exec.Command()`?

Reply:
Or even better is to put a already corrupted key in the `e2e/key/testdata`?... <- I vote for this.